app/services/UploadService.scala

```diff
@@ -0,0 +1,42 @@
+package services
+
+import com.nimbusds.oauth2.sdk.token.BearerAccessToken
+import configuration.GraphQLConfiguration
+import graphql.codegen.StartUpload.{startUpload => su}
+import graphql.codegen.AddFilesAndMetadata.{addFilesAndMetadata => afam}
+import graphql.codegen.UpdateConsignmentStatus.{updateConsignmentStatus => ucs}
+import graphql.codegen.types.{AddFileAndMetadataInput, ConsignmentStatusInput, StartUploadInput}
+import services.ApiErrorHandling.sendApiRequest
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class UploadService @Inject()(val graphqlConfiguration: GraphQLConfiguration)(implicit val ec: ExecutionContext) {
+  private val startUploadClient = graphqlConfiguration.getClient[su.Data, su.Variables]()
+  private val addFilesAndMetadataClient = graphqlConfiguration.getClient[afam.Data, afam.Variables]()
+  private val updateConsignmentStatusClient = graphqlConfiguration.getClient[ucs.Data, ucs.Variables]()
+
+  def updateConsignmentStatus(consignmentStatusInput: ConsignmentStatusInput, token: BearerAccessToken): Future[Int] = {
+    val variables = ucs.Variables(consignmentStatusInput)
+    sendApiRequest(updateConsignmentStatusClient, ucs.document, token, variables).map(data => {
+      data.updateConsignmentStatus match {
+        case Some(response) => response
+        case None => throw new RuntimeException(s"No data returned when updating the consignment status for ${consignmentStatusInput.consignmentId}")
+      }
+    })
+  }
+
+  def startUpload(startUploadInput: StartUploadInput, token: BearerAccessToken): Future[String] = {
```

Comment:
Is it possible to have no response here? If so, should an error be thrown?

Reply:
The `ApiErrorHandling` object calls `result.data.get` so if data was optional, we'd get a (slightly ugly) error and it also throws a `GraphQlException` if the API populates the errors field. I think that covers everything that could happen. What do you think?